### ISSUE TITLE
new site

### ISSUE THREAD

**Issue by aaditya-029** [aman77singh/Aman-Singh#6]

**Comment by aaditya-029** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Quantum Anomaly Detection</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            background-color: #f4f4f4;
            margin: 0;
            padding: 0;
        }
        .container {
            width: 80%;
            margin: 0 auto;
            padding: 20px;
            background-color: #fff;
            box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
        }
        h1 {
            text-align: center;
            color: #333;
        }
        .form-group {
            margin: 20px 0;
        }
        label {
            display: block;
            margin-bottom: 5px;
        }
        input[type="file"],
        input[type="number"] {
            width: 100%;
            padding: 10px;
            margin-bottom: 10px;
        }
        button {
            padding: 10px 15px;
            background-color: #007bff;
            color: #fff;
            border: none;
            cursor: pointer;
            width: 100%;
        }
        button:hover {
            background-color: #0056b3;
        }
        #output {
            margin-top: 20px;
            padding: 10px;
            background-color: #e9ecef;
            color: #333;
        }
    </style>
</head>
<body>
    <div class="container">
        <h1>Quantum Anomaly Detection</h1>
        <form id="uploadForm" enctype="multipart/form-data">
            <div class="form-group">
                <label for="file">Upload CSV File</label>
                <input type="file" id="file" name="file" accept=".csv">
            </div>
            <div class="form-group">
                <label for="epochs">Number of Epochs</label>
                <input type="number" id="epochs" name="epochs" value="50" min="1">
            </div>
            <button type="button" onclick="submitForm()">Train Model</button>
        </form>
        <div id="output"></div>
    </div>

    <script>
        function submitForm() {
            const form = document.getElementById('uploadForm');
            const formData = new FormData(form);
            const outputDiv = document.getElementById('output');

            fetch('/train', {
                method: 'POST',
                body: formData,
            })
            .then(response => response.json())
            .then(data => {
                outputDiv.innerHTML = `<h3>Training Complete</h3><pre>${JSON.stringify(data, null, 2)}</pre>`;
            })
            .catch(error => {
                outputDiv.innerHTML = `<h3>Error</h3><p>${error}</p>`;
            });
        }
    </script>
</body>
</html>

**Comment by aaditya-029** [comment]
from flask import Flask, request, jsonify
import numpy as np
import pandas as pd
import pennylane as qml
from sklearn.model_selection import train_test_split
from sklearn.decomposition import PCA

app = Flask(__name__)

# Load the global device parameters
global dev_params

# Define the quantum device and circuit as shown in your original code
# Define the loss and training functions as before

@app.route('/train', methods=['POST'])
def train():
    # Get the uploaded CSV file and the number of epochs from the request
    file = request.files['file']
    epochs = int(request.form['epochs'])

    # Load dataset
    data = pd.read_csv(file)
    X = data.iloc[:, :-1].values
    y = data.iloc[:, -1].values

    # Split the dataset
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

    # Reduce dimensionality if necessary (based on your backend code)
    num_qubits = min(X_train.shape[1], 4)
    if X_train.shape[1] > num_qubits:
        pca = PCA(n_components=num_qubits)
        X_train = pca.fit_transform(X_train)
        X_test = pca.transform(X_test)

    # Train the quantum model
    train_quantum_model(X_train, y_train, epochs)

    # Evaluate the model
    accuracy, report = evaluate_model(X_test, y_test)

    # Return the result as JSON
    return jsonify({"accuracy": accuracy, "report": report})

if __name__ == '__main__':
    app.run(debug=True)
